pkg/cmd/admin/router/router.go

```diff
@@ -402,49 +408,37 @@ func generateSecretsConfig(cfg *RouterConfig, namespace string, defaultCert []by
 	return secrets, volumes, mounts, nil
 }
 
-func generateProbeConfigForRouter(cfg *RouterConfig, ports []kapi.ContainerPort) *kapi.Probe {
-	var probe *kapi.Probe
-
-	if cfg.Type == "haproxy-router" {
-		probe = &kapi.Probe{}
-		healthzPort := defaultHealthzPort
-		if cfg.StatsPort > 0 {
-			healthzPort = cfg.StatsPort
-		}
+func generateProbeConfigForRouter(cfg *RouterConfig, ports []kapi.ContainerPort, probePath string, initialDelay int32) *kapi.Probe {
+	probe := &kapi.Probe{InitialDelaySeconds: initialDelay}
+	probePort := defaultProbePort
+	if cfg.ProbePort > 0 {
+		probePort = cfg.ProbePort
+	}
 
-		probe.Handler.HTTPGet = &kapi.HTTPGetAction{
-			Path: "/healthz",
-			Port: intstr.IntOrString{
-				Type:   intstr.Int,
-				IntVal: int32(healthzPort),
-			},
-		}
+	probe.Handler.HTTPGet = &kapi.HTTPGetAction{
+		Path: probePath,
+		Port: intstr.IntOrString{
+			Type:   intstr.Int,
+			IntVal: int32(probePort),
+		},
+	}
 
-		// Workaround for misconfigured environments where the Node's InternalIP is
-		// physically present on the Node.  In those environments the probes will
-		// fail unless a host firewall port is opened
-		if cfg.HostNetwork {
-			probe.Handler.HTTPGet.Host = "localhost"
-		}
+	// Workaround for misconfigured environments where the Node's InternalIP is
+	// physically present on the Node.  In those environments the probes will
+	// fail unless a host firewall port is opened
+	if cfg.HostNetwork {
+		probe.Handler.HTTPGet.Host = "localhost"
 	}
 
 	return probe
 }
 
 func generateLivenessProbeConfig(cfg *RouterConfig, ports []kapi.ContainerPort) *kapi.Probe {
-	probe := generateProbeConfigForRouter(cfg, ports)
-	if probe != nil {
-		probe.InitialDelaySeconds = 10
-	}
-	return probe
+	return generateProbeConfigForRouter(cfg, ports, "/alive", 10)
```

Comment:
This is just a default thing that has been used for a while.
If not good for some users, further customization can be through editing the router's DC.